src/PlusServer/Commands/vtkPlusCapistranoCommand.cxx

```diff
@@ -0,0 +1,339 @@
+  {
+    return PLUS_FAIL;
+  }
+  XML_WRITE_STRING_ATTRIBUTE_IF_NOT_EMPTY(DeviceId, aConfig);
+  XML_WRITE_STRING_ATTRIBUTE_IF_NOT_EMPTY(CommandName, aConfig);
+  XML_WRITE_STRING_ATTRIBUTE_IF_NOT_EMPTY(CommandValue, aConfig);
+  return PLUS_SUCCESS;
+}
+
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusCapistranoCommand::Execute()
+{
+  LOG_DEBUG("vtkPlusSetUsParameterCommand::Execute: " << (!this->Name.empty() ? this->Name : "(undefined)")
+            << ", device: " << (this->DeviceId.empty() ? "(undefined)" : this->DeviceId));
+
+  if (this->Name.empty())
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", "No command name specified.");
+    return PLUS_FAIL;
+  }
+  else if (!igsioCommon::IsEqualInsensitive(this->Name, CAPISTRANO_COMMAND))
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", "Unknown command name: " + this->Name + ".");
+    return PLUS_FAIL;
+  }
+  vtkPlusDataCollector* dataCollector = GetDataCollector();
+  vtkPlusDevice* aDevice = NULL;
+  if (dataCollector->GetDevice(aDevice, this->DeviceId) != PLUS_SUCCESS)
+  {
+    this->QueueCommandResponse(PLUS_FAIL, "Command failed. See error message.", std::string("Device ")
+                               + (this->DeviceId.empty() ? "(undefined)" : this->DeviceId) + std::string(" is not found."));
+    return PLUS_FAIL;
+  }
+
+  std::string DeviceId = (aDevice->GetDeviceId().empty() ? "(unknown)" : aDevice->GetDeviceId());
+  std::string resultString = "<CommandReply>";
+  std::string error = "";
+  std::map < std::string, std::pair<IANA_ENCODING_TYPE, std::string> > metaData;
+  PlusStatus status = PLUS_SUCCESS;
+
+  std::list<std::pair<std::string, std::string>>::iterator paramIt;
+  for (paramIt = this->RequestedParameterChanges.begin(); paramIt != this->RequestedParameterChanges.end(); ++paramIt)
+  {
+    std::string parameterName = paramIt->first;
+    std::string value = paramIt->second;
+    resultString += "<Parameter Name=\"" + parameterName + "\"";
+
+    if (parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_WOBBLE_RATE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_GET_WOBBLE_RATE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_MIS_MODE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_MIS_PULSE_PERIOD
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_GET_JITTER_COMPENSATION
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_JITTER_COMPENSATION
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_FREEZE_PROBE
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_IS_PROBE_FROZEN
+        && parameterName != vtkPlusCapistranoVideoSource::CAPISTRANO_SET_BIDIRECTIONAL_MODE)
+    {
+      error += "Invalid parameter " + parameterName + ". ";
```

Comment:
For some reason these error messages don't get propagated to the CommandReply? This is despite the later line `commandResponse->SetErrorString(error);`. I'm not sure why this is, SetUsParameter seems to have similar behavior

![image](https://user-images.githubusercontent.com/3487395/215906183-f316cd62-7f46-4b34-b440-5c9779d88c63.png)

Reply:
I think the error message shows up in the second response window (for both Set/GetUsParameter and CapsitranoCommand). Is this not as expected?
![image](https://user-images.githubusercontent.com/88200986/216079289-c2430cd4-c4ea-4f3a-9b57-0fad12d919da.png)

Reply:
Oh I see, I was confused as there are some error responses from the GenericSerial command class that get passed through the CommandReply XML string. I suppose this is fine.